crates/project-model/src/workspace.rs

```diff
@@ -150,13 +150,14 @@ impl ProjectWorkspace {
     ) -> Result<ProjectWorkspace> {
         let res = match manifest {
             ProjectManifest::ProjectJson(project_json) => {
+                let project_json = project_json.canonicalize()?;
                 let file = fs::read_to_string(&project_json).with_context(|| {
                     format!("Failed to read json file {}", project_json.display())
                 })?;
                 let data = serde_json::from_str(&file).with_context(|| {
                     format!("Failed to deserialize json file {}", project_json.display())
                 })?;
-                let project_location = project_json.parent().to_path_buf();
+                let project_location = project_json.parent().unwrap().to_path_buf();
```

Comment:
Could potentially move the canonicalization to `ManifestPath` to avoid the `unwrap()` here, or do the canonicalization in every case inside `ProjectJson::new()`, or do the canonicalization already when creating the `ProjectManifest`. Many options :)

Reply:
Or alternatively the `rust-project.json` could have absolute paths generated by meson for `root_module` and others but not sure that's allowed?

Reply:
Absolute paths should work fine for `root_module` and friends. Not sure if canonicalizing is the right behavior we want here or not.

Reply:
As relative paths can be used in `rust-project.json` it would make sense to at least resolve any symlinks for getting the real path of the file.

I'll also submit a PR to meson for writing absolute paths in there though.

Reply:
Wouldn't resolving symlinks when loading rust-project.json break when the editor doesn't resolve symlinks before making lsp requests?

Reply:
I don't know how that works in detail, but with nvim/coc it works fine this way